src/core/ext/filters/client_channel/client_channel.cc

```diff
@@ -3057,6 +3070,32 @@ void CallData::InjectRecvInitialMetadataReadyForConfigSelectorCommitCallback(
       &recv_initial_metadata_ready_;
 }
 
+void CallData::RecvTrailingMetadataReadyForAdditionalErrorContext(
```

Comment:
I don't think it's necessary to intercept the recv_trailing_metadata op to do this.  Instead, you can just modify the error here:

https://github.com/grpc/grpc/blob/5b5e3f84a195929a1774f1dc69e6e8c05caa7817/src/core/ext/filters/client_channel/client_channel.cc#L2934

Reply:
If we modify the error in the resolver queue canceller, then we'll only add "occurred_while_awaiting_name_resolution" if the call times out or is cancelled while name resolution is still in progress. But if name resolution finishes with an error, then we won't add this detail.

This is not a huge problem, but creates a small inconsistency. WDYT?

Reply:
In the case where the resolver returns an error and the call is not wait_for_ready, we're already using the resolver error to fail the call:

https://github.com/grpc/grpc/blob/5b5e3f84a195929a1774f1dc69e6e8c05caa7817/src/core/ext/filters/client_channel/client_channel.cc#L3139

If you want to augment that error message, go for it, but since we already have a place in the code where we're generating this error, we might as well just change it to include what we want instead of taking a separate callback to change it again later.

Reply:
It looks like a problem with augmenting the error in the resolver queue canceller is that it only affects the final RPC `error` object if there is a RECV_STATUS op pending at the time that the RPC is ejected from the resolver queue.

I've brought down the new test cases from https://github.com/grpc/grpc/pull/25158 into this PR, and adjusted the RPC to use a bidi streaming RPC, to illustrate this issue.

Specifically, I want to make sure that name resolution debug context is visible in the following case:
 - someone is using a streaming RPC
 - they start out by trying to do an initial write, which fails or times out while waiting for name resolution to complete